crates/sui-keys/src/keystore.rs

```diff
@@ -153,48 +173,152 @@ impl AccountKeystore for FileBasedKeystore {
+    fn create_alias_if_not_exists(&self, alias: Option<String>) -> Result<String, anyhow::Error> {
+        let existing_aliases = self
+            .aliases
+            .clone()
+            .into_values()
+            .map(|x| x.alias)
+            .collect::<HashSet<_>>();
+        match alias {
+            Some(a) if existing_aliases.contains(&a) => {
+                bail!("Alias {a} already exists. Please choose another alias.")
+            }
+            Some(a) => Ok(a),
+            None => Ok(random_name(&existing_aliases)),
+        }
+    }
 }
 
 impl FileBasedKeystore {
     pub fn new(path: &PathBuf) -> Result<Self, anyhow::Error> {
         let keys = if path.exists() {
-            let reader = BufReader::new(
-                File::open(path)
-                    .map_err(|e| anyhow!("Can't open FileBasedKeystore from {:?}: {e}", path))?,
-            );
-            let kp_strings: Vec<String> = serde_json::from_reader(reader)
-                .map_err(|e| anyhow!("Can't deserialize FileBasedKeystore from {:?}: {e}", path))?;
+            let reader =
+                BufReader::new(File::open(path).with_context(|| {
+                    format!("Cannot open the keystore file: {}", path.display())
+                })?);
+            let kp_strings: Vec<String> = serde_json::from_reader(reader).with_context(|| {
+                format!("Cannot deserialize the keystore file: {}", path.display(),)
+            })?;
             kp_strings
                 .iter()
                 .map(|kpstr| {
                     let key = SuiKeyPair::decode_base64(kpstr);
                     key.map(|k| (Into::<SuiAddress>::into(&k.public()), k))
                 })
                 .collect::<Result<BTreeMap<_, _>, _>>()
-                .map_err(|e| anyhow::anyhow!("Invalid Keypair file {:#?} {:?}", e, path))?
+                .map_err(|e| anyhow!("Invalid keystore file: {}. {}", path.display(), e))?
         } else {
             BTreeMap::new()
         };
 
+        // check aliases
+        let mut aliases_path = path.clone();
+        aliases_path.set_extension("aliases");
+
+        let aliases = if aliases_path.exists() {
+            let reader = BufReader::new(File::open(&aliases_path).with_context(|| {
+                format!(
+                    "Cannot open aliases file in keystore: {}",
+                    aliases_path.display()
+                )
+            })?);
+
+            let aliases: Vec<Alias> = serde_json::from_reader(reader).with_context(|| {
+                format!(
+                    "Cannot deserialize aliases file in keystore: {}",
+                    aliases_path.display(),
+                )
+            })?;
+
+            aliases
+                .into_iter()
+                .map(|alias| {
+                    let key = SuiKeyPair::decode_base64(&alias.public_key_base64);
+                    key.map(|k| (Into::<SuiAddress>::into(&k.public()), alias))
```

Comment:
I think you can probably use `SuiAddress::from` instead of  this?

```suggestion
                    key.map(|k| (SuiAddress::from(&k.public()), alias))
```